test/test_jit_cuda_fuser.py

```diff
@@ -649,6 +650,127 @@ def test_reduction(self):
                         perm1 = range(len(x))
                         self._reduction_helper(x, axes, torch.float32, "cuda", perm0, perm1, keepdim)
 
+    def _layer_norm_helper(self, shape, norm_shape, dtype, device):
+        class MyLayerNorm(torch.nn.Module):
+            __constants__ = ['norm_shape']
+
+            def __init__(self):
+                super(MyLayerNorm, self).__init__()
+                self.norm_shape = norm_shape
+
+            def forward(self, x: torch.Tensor, y: torch.Tensor):
+                o = torch.add(x, y)
+                o = torch.nn.functional.layer_norm(o, self.norm_shape)
+                return o
+
+        t = MyLayerNorm()
+
+        x = torch.randn(shape, dtype=dtype, device=device)
+        y = torch.randn(shape, dtype=dtype, device=device)
+        t_jit = torch.jit.script(t)
+        jit_o = t_jit(x, y)
+        jit_o = t_jit(x, y)
+        o = t(x, y)
+        self.assertEqual(o.dtype, jit_o.dtype)
+        # numerical issues here due to our scheduling.
+        # can't use `self.assertEqual(o, jit_o)`
+        self.assertTrue(self._compare("comparing output failed", o, jit_o, 1e-4))
+        self.assertGraphContains(t_jit.graph_for(x, y), FUSION_GUARD)
+
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_layer_norm(self):
+        dims = 4
+        rnds = 3
+        for idx in range(rnds):
+            for offset in range(1, dims):
+                input_shape = [random.randint(30, 100) for idx in range(dims)]
+                norm_shape = [input_shape[idx] for idx in range(dims - offset, dims)]
+                self._layer_norm_helper(input_shape, norm_shape, torch.float32, "cuda")
+
+    def _batch_norm_helper(self, shape, dtype, device):
+        class MyBatchNorm(torch.nn.Module):
+            def __init__(self):
+                super(MyBatchNorm, self).__init__()
+
+            def forward(self, x: torch.Tensor, y: torch.Tensor, r_mean : torch.Tensor, r_var : torch.Tensor):
+                o = torch.add(x, y)
+                o = torch.nn.functional.batch_norm(o, r_mean, r_var, training=True)
+                return o
+
+        t = MyBatchNorm()
+
+        x = torch.randn(shape, dtype=dtype, device=device)
+        y = torch.randn(shape, dtype=dtype, device=device)
+        running_mean = torch.randn(shape[1], dtype=dtype, device=device)
+        running_var = torch.randn(shape[1], dtype=dtype, device=device)
+        t_jit = torch.jit.script(t)
+        jit_o = t_jit(x, y, running_mean, running_var)
```

Comment:
nitpick: I know we don't have `running_mean` / `running_var` updated yet, but we could still put the right code here. Each iteration should have their own clone of `running_mean` / `running_var` and put a comment that we should validate those results once we have the implementation.

Reply:
:+1: